src/neo4j_graphrag/llm/base.py

```diff
@@ -52,6 +54,21 @@ def invoke(self, input: str) -> LLMResponse:
             LLMGenerationError: If anything goes wrong.
         """
 
+    @abstractmethod
+    def chat(self, input: str, chat_history: list[str]) -> LLMResponse:
```

Comment:
I feel like it could be done in one method, with an optional `chat_history`. The difference would be dealt with in a `get_messages` method, that would do something like this:

```
def get_messages(self, input, chat_history) -> list:
    messages = []
    if self.system_instructions:
        messages.append({"role": "system", "content": self.system_instructions}})
    if chat_history:
        # handle chat history messages format
        messages.extend(chat_history)
    messages.append({"role": "user", "content": input})
    return messages
```

This is an advantage if most of the LLM providers share the same format for messages, in which case we can implement this method only once in the interface.

Reply:
Ok, that works.